### PR TITLE
feat(popover): add new popover component

### PR DESCRIPTION
**Pull request by fpt-duongtx**
### To support user to show menu
* Add new Popover component
* Add document for it

### CODE REVIEW

File: packages/react-compass/src/popover/popover.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React, {useCallback} from 'react'
+import {StyledComponentProps} from '../utils/stitches.types'
+import {useDOMRef} from '../utils/use-dom-ref'
+import {StyledPopover, StyledPopoverContent} from './popover.style'
+
+export type PopoverDirection =
+  | 'bottom-right'
+  | 'bottom-left'
+  | 'top-right'
+  | 'top-left'
```
Comment (thaycacac) [inline packages/react-compass/src/popover/popover.tsx:10]: I think need add more `bottom-center` and `top-center` for this type

File: packages/react-compass/src/popover/popover.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      className='cdg-popover'
+      ref={popoverRef}
+      css={css}
+      {...delegated}
+      onBlur={handleWrapperBlur}
+      tabIndex={0}
+    >
+      {anchor}
+      {isOpen && (
+        <StyledPopoverContent className={`cdg-popover-content ${direction}`}>
```
Comment (thaycacac) [inline packages/react-compass/src/popover/popover.tsx:62]: I think need to handle auto display position. For example, I use popover component and passed with the `top-right` position to prop, but when the button is at the top of the page, it should automatically display in the bottom-right position.

Demo: https://react-spectrum.adobe.com/react-spectrum/Menu.html

Comment (fpt-duongtx) [reply packages/react-compass/src/popover/popover.tsx:64]: Hi @thaycacac ,

Thanks for your advice here!
I think our components are using popover, they are on a fixed position. However, to handle the position dynamically, we will need to spend a lot of effort. So I would not support it for now.
Could you please help to convert it to a ticket then we can support more features for Popover later?

Comment (thaycacac) [reply packages/react-compass/src/popover/popover.tsx:64]: ok. thanks

**Review by thaycacac — APPROVED** [review]

**Pull request merged**